src/main.ts

```diff
@@ -231,6 +247,19 @@ function animate(): void {
         }
       }
 
+      if (isPlayer(object)) {
+        for (const mineral of minerals) {
+          if (collision(object, mineral)) {
+            // push player back
+            const direction = new THREE.Vector3();
+            direction.subVectors(mineral.position, object.position);
+            direction.normalize();
+            direction.multiplyScalar(0.1);
+            object.position.sub(direction);
+          }
+        }
```

Comment:
This is very costy and ineffizient
Maybe there is a way to only check minerals near to the player
Or maybe only check with visible (inside frustum)